### PR TITLE
Add Breaking To Ring Of Loki

### PR DESCRIPTION
**Pull request by LewisSaber**
Allows Player to break blocks with ring of loki, added keybind to toggle breaking on and off

### CODE REVIEW

File: src/main/java/vazkii/botania/common/item/relic/ItemLokiRing.java [hunk 1]
```diff
@@ -170,43 +194,78 @@ public void onPlayerInteract(PlayerInteractEvent event) {
...
 	}
+
+	public static void setBreakingMode(ItemStack stack, boolean state) {
+		stack.stackTagCompound.setBoolean(TAG_BREAKING_MODE, state);
+	}
+
+
 	@SideOnly(Side.CLIENT)
-	public static void renderHUDNotification(){
+	public static void renderHUDNotification(String type){
```
Comment (combusterf) [inline src/main/java/vazkii/botania/common/item/relic/ItemLokiRing.java:208]: Coding antipattern: use enums.

File: src/main/java/vazkii/botania/common/item/relic/ItemLokiRing.java [hunk 2]
```diff
@@ -61,13 +66,32 @@ public class ItemLokiRing extends ItemRelicBauble implements IExtendedWireframeC
...
 
 	public ItemLokiRing() {
 		super(LibItemNames.LOKI_RING);
 		MinecraftForge.EVENT_BUS.register(this);
 	}
 
+	@SubscribeEvent
+    public void onBlockBreak(BlockEvent.BreakEvent event) {
+        EntityPlayer player = event.getPlayer();
+
```
Comment (combusterf) [inline src/main/java/vazkii/botania/common/item/relic/ItemLokiRing.java:80]: Several indentation issues in this method, please correct them.

File: src/main/java/vazkii/botania/common/item/relic/ItemLokiRing.java [hunk 3]
```diff
@@ -170,43 +194,78 @@ public void onPlayerInteract(PlayerInteractEvent event) {
...
+		return stack.stackTagCompound.getBoolean(TAG_BREAKING_MODE);
+	}
+
 	public static void breakOnAllCursors(EntityPlayer player, Item item, ItemStack stack, int x, int y, int z, int side) {
 		ItemStack lokiRing = getLokiRing(player);
-		if(lokiRing == null || player.worldObj.isRemote || !(item instanceof ISequentialBreaker))
+		if(lokiRing == null || player.worldObj.isRemote || !isRingEnabled(lokiRing) || !isRingBreakingEnabled(lokiRing))
 			return;
-
+		boolean isISequentalBreaker = item instanceof ISequentialBreaker;
```
Comment (combusterf) [inline src/main/java/vazkii/botania/common/item/relic/ItemLokiRing.java:250]: Variable can be removed; testing for `breaker != null` instead makes the code shorter and the nullability correctness easier to demonstrate.

File: src/main/java/vazkii/botania/common/item/relic/ItemLokiRing.java [hunk 4]
```diff
@@ -170,43 +194,78 @@ public void onPlayerInteract(PlayerInteractEvent event) {
...
 		List<ChunkCoordinates> cursors = getCursorList(lokiRing);
-		ISequentialBreaker breaker = (ISequentialBreaker) item;
+		ISequentialBreaker breaker  = null;
+		if(isISequentalBreaker)
+		breaker = (ISequentialBreaker) item;
```
Comment (combusterf) [inline src/main/java/vazkii/botania/common/item/relic/ItemLokiRing.java:254]: Please fix indentation in this method

File: src/main/java/vazkii/botania/common/network/PacketHandler.java [hunk 5]
```diff
@@ -13,7 +13,9 @@ private PacketHandler() {}
             NetworkRegistry.INSTANCE.newSimpleChannel(LibMisc.MOD_ID.toLowerCase());
 
     public static void initPackets() {
-        INSTANCE.registerMessage(PacketLokiToggle.class, PacketLokiToggle.class, 1, Side.SERVER);
+       INSTANCE.registerMessage(PacketLokiToggle.class, PacketLokiToggle.class, 1, Side.SERVER);
```
Comment (combusterf) [inline src/main/java/vazkii/botania/common/network/PacketHandler.java:16]: Please fix indentation

**Review by combusterf — CHANGES_REQUESTED** [review]

File: src/main/java/vazkii/botania/common/item/relic/ItemLokiRing.java (hunk 2 above)
Comment (combusterf) [reply src/main/java/vazkii/botania/common/item/relic/ItemLokiRing.java]: Almost good, except the method is now unaligned with the rest of the file (and the annotation).

Comment (LewisSaber) [reply src/main/java/vazkii/botania/common/item/relic/ItemLokiRing.java]: It looked aligned in my editor, fixed

**Review by combusterf — APPROVED** [review]

**Pull request merged**